plugins/oauth/server/providers/globus.py

```diff
@@ -0,0 +1,149 @@
+                    'openid', 'profile', 'email')
+    _TOKEN_URL = 'https://auth.globus.org/v2/oauth2/token'
+    _JWK_KEY_URL = 'https://auth.globus.org/jwk.json'
+    jwk_keys = None
+
+    def getClientIdSetting(self):
+        return self.model('setting').get(
+            constants.PluginSettings.GLOBUS_CLIENT_ID)
+
+    def getClientSecretSetting(self):
+        return self.model('setting').get(
+            constants.PluginSettings.GLOBUS_CLIENT_SECRET)
+
+    @classmethod
+    def getUrl(cls, state):
+        clientId = cls.model('setting').get(
+            constants.PluginSettings.GLOBUS_CLIENT_ID)
+
+        if clientId is None:
+            raise Exception('No Globus client ID setting is present.')
+
+        callbackUrl = '/'.join((getApiUrl(), 'oauth', 'globus', 'callback'))
+
+        query = urllib.parse.urlencode({
+            'response_type': 'code',
+            'access_type': 'online',
+            'client_id': clientId,
+            'redirect_uri': callbackUrl,
+            'state': state,
+            'scope': ' '.join(cls._AUTH_SCOPES)
+        })
+        return '%s?%s' % (cls._AUTH_URL, query)
+
+    def getToken(self, code):
+        params = {
+            'grant_type': 'authorization_code',
+            'code': code,
+            'client_id': self.clientId,
+            'client_secret': self.clientSecret,
+            'redirect_uri': self.redirectUri
+        }
+        resp = self._getJson(method='POST', url=self._TOKEN_URL,
+                             data=params,
+                             headers={'Accept': 'application/json'})
+        if 'error' in resp:
+            raise RestException(
+                'Got an error exchanging token from provider: "%s".' % resp,
+                code=502)
+        return resp
+
+    def getUser(self, token):
+        id_token = token.get('id_token')
+
+        if self.jwk_keys is None:
+            # Get public keys required for decoding 'id_token'
+            self.jwk_keys = self._getJson(method='GET', url=self._JWK_KEY_URL)
+
+        # There should be only one entry
+        keyobj = self.jwk_keys['keys'][0]
+
+        # Create RSA key from JSON spec
+        key = rsa.RSAPublicNumbers(
+            n=decode_value(keyobj['n']),
+            e=decode_value(keyobj['e'])
+        ).public_key(default_backend())
+
+        # Decode 'id_token'
+        # BEWARE: leeway should be 0, but upstream server seems to be desynch
+        identity = jwt.decode(
+            id_token, key, algorithm='RS512', leeway=100,
+            audience=self.clientId)
+
+        oauthId = identity.get('sub')
+        if not oauthId:
+            raise RestException(
+                'Globus identity did not return a valid ID.', code=502)
+
+        email = identity.get('email')
+        if not email:
+            raise RestException(
+                'Globus identity did not return a valid email.', code=502)
+
+        name = identity.get('name').split()
```

Comment:
This will only work if `identity` actually contains a `'name'` key. Otherwise, we are attempting to split `None`, which will throw a confusing error message. We should either

* Change to use `identity['name'].split()` if we are sure we will be given a name.
* Change to `identity.get('name', some_default_string).split()`
* If there is no name field or an empty name field, we should raise a more useful exception message.